src/harness/test_harness_objects.py

```diff
@@ -0,0 +1,324 @@
+    self.registration_request = registration_request
+
+  def getCbsdId(self):
+    return self.cbsd_id
+
+  def getGrantObject(self, grant_id):
+      return self.grant_objects[grant_id]
+
+  def getRegistrationRequest(self):
+    return self.registration_request
+
+  def hasActiveGrant(self):
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isActive():
+        return True
+    return False
+
+  def hasAuthorizedGrant(self):
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isGrantAuthorizedInLastHeartbeat():
+        return True
+    return False
+
+  def constructHeartbeatRequestForAllActiveGrants(self):
+    """Construct list of heartbeat requests of all active grants."""
+    heartbeat_requests = []
+    for grant_object in self.grant_objects.itervalues():
+      if grant_object.isActive():
+        heartbeat_request = grant_object.constructHeartbeatRequest()
+        heartbeat_requests.append(heartbeat_request)
+    return heartbeat_requests
+
+  def getOperationParamsOfAllAuthorizedGrants(self):
+    """Returns the list of operation params of all authorized grants."""
+    operation_params = []
+    for grant_object in self.grant_objects.values():
+      if grant_object.isGrantAuthorizedInLastHeartbeat():
+        operation_param = grant_object.getRequestOperationParam()
+        operation_params.append(operation_param)
+    return operation_params
+
+
+class DomainProxy(object):
+  """Performs Domain Proxy related operations.
+
+  This class stores Domain Proxy related information like the SSL certificate, key to be used and the set of
+  CBSD objects belonging to this Domain Proxy. Bulk operations like registration, grant and heartbeat
+  procedures are performed on all of the CBSDs belonging to this Domain Proxy.
+  """
+  def __init__(self, ssl_cert, ssl_key, testcase):
+    """
+    Args:
+      ssl_cert: Path to SSL cert file.
+      ssl_key: Path to SSL key file.
+      testcase: test case object from the caller.
+    """
+    self.ssl_cert = ssl_cert
+    self.ssl_key = ssl_key
+    self.cbsd_objects = {}
+    self.testcase = testcase
+
+  def addCbsdsAndGrants(self, registration_requests, grant_requests):
+    """Construct CBSD object based on the the result of registration and
+    grant requests.
+
+    Args
+      registration_requests: A list of dictionary elements ,where each list elements is
+        individual CBSD registration requests (each of which is itself a dictionary).
+      grant_requests: A list of dictionary elements,where each list elements is
+        individual CBSD grant requests (each of which is itself a dictionary).
+    """
+    # Checking if the number of registration requests matches number of grant requests
+    # There should be exactly one grant request per registration request.
+    assert len(grant_requests) == len(registration_requests)
+    cbsd_ids = self.testcase.assertRegistered(registration_requests)
```

Comment:
This doesn't use the domain proxy certificate, please modify the assertRegistered function to allow an optional certificate and key then pass that here.

Reply:
Changed   as per suggestion.